Feature/frontend

tsureshkumar opened the pull request:
Modern Frontend for Tamil Wordle

A simple [Next Js](https://nextjs.org/) based front end

- uses React as UI component framework
- uses Tailwind css for styling 
- adaptive UI based on screen size
- easy to deploy on Varcel's hobby project (check [this game deployed there](https://tamil-wordle.vercel.app/))

Functionalities
- checks for duplicate words
- color coding for jumbled/correct/invalid letters
- on success, displays text based emoji tiles for easy sharing

Further Plans
- show intuitive dials for showing which letters have been tried and which can be tried now
- on screen keyboard for those who have not installed system level Tamil keyboard

Commit:
enable cors for frontend

- fetch calls from react needs to be CORS enabled

tsureshkumar committed:
allow trusted sites in CORS

- remove wildcard
- parses the origin header and sends CORS only if from allowed domains

- uses a map lookup and if present, send to be callable from remote origin

tsureshkumar committed:
add .gitignore

tsureshkumar committed:
initial next js based front end

psankar commented on frontend/tamil-wordle/messages-ta.js:
```diff
@@ -0,0 +1,7 @@
+export const IntlMsg = {
+    game_name: "தமிழ் வார்த்தை விளையாட்டு (Tamil Wordle) ",
```
சொல் ?

வார்த்தை வடமொழி

psankar commented on frontend/tamil-wordle/messages-ta.js:
```diff
@@ -0,0 +1,7 @@
+    btn_try: "சரி பார்க்க",
+    btn_game_over: "கெலிச்சாச்சு! பகிரவும்!",
+    msg_invalid_length: "*சரியான எண்ணிக்கையிலான எழுத்துக்களை இடவும்",
+    msg_duplicate: "*ஏற்கனவே முயற்சித்தது!",
```
முயன்றது ?

முயற்சித்தது தவறான சொல்.

tsureshkumar committed:
fix sharing transparency issue

some share the screeshot of result screen which actually
reveals the answers.

This removes the transparency of the overlay screen and make it opaque

tsureshkumar committed:
add copy/share button

makes it easy for users to directly click and share.

Share WebAPI is not yet widely avaialble. As a fallback,
the text is copied to clipboard if not available.

tsureshkumar committed:
fix surrounding spaces

- remove spaces before validating for length

tsureshkumar committed:
show errors on top

- add capability show errors, info and warning through a new react component
- show error if any communication error happens

tsureshkumar committed:
show hint while typing

- it is difficult to remember what latters have been tried earlier
  and it's correct position. This now shows a hint tile below the input
  box with color coding and animation to help you not *repeat* a
  mistake.

- this introduces game state with the previous letters tried and its
  result. The memory could increase with many tries, optimizations could
  be considered latter to reduce the unnecessary space.

tsureshkumar committed:
make use of full height

- currently, this was using only a fixed height of 56 points.  Now it
  utilizes the full screen

- increase the font for better visibility

tsureshkumar committed:
add tamil letters

tsureshkumar committed:
fix issue of sometime adding duplicate tries on ENTER

Sometimes, on pressing ENTER key in text box, the duplicate checking
is not working. This is becase, the word is added to the list only after
a successful network call completion.  If there is more network delay,
subsequent checkDuplicate returns true and further queues up the calls.

To fix it, separate data structure is kept to check for duplicates alone.
Any word that has been already processed or in processing in network, will
be checked against it and udpated immediately.

tsureshkumar committed:
update gitignore
